src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -226,266 +226,189 @@ public void initForSectionPanelViews(InstructorAttributes instructor,
             break;
         case GIVER_RECIPIENT_QUESTION:
             Map<String, Map<String, List<FeedbackResponseAttributes>>> sortedResponsesForGrq =
                     bundle.getResponsesSortedByGiverRecipientQuestion(isGroupedByTeam());
 
-            buildSectionPanelsForViewByParticipantParticipantQuestion(sortedResponsesForGrq, viewType.additionalInfoId());
+            buildSectionPanelForViewByParticipantParticipantQuestion(selectedSection,
+                    sortedResponsesForGrq, viewType.additionalInfoId());
             break;
         default:
             Assumption.fail();
             break;
         }
-        
+
     }
-    
-    private void buildSectionPanelsForViewByParticipantParticipantQuestion(
-                         Map<String, Map<String, List<FeedbackResponseAttributes>>> sortedResponses,
-                         String additionalInfoId) {
+
+    private void buildSectionPanelForViewByParticipantParticipantQuestion(
+                                    String section,
+                                    Map<String, Map<String, List<FeedbackResponseAttributes>>> sortedResponses,
+                                    String additionalInfoId) {
         sectionPanels = new LinkedHashMap<String, InstructorFeedbackResultsSectionPanel>();
         InstructorFeedbackResultsSectionPanel sectionPanel = new InstructorFeedbackResultsSectionPanel();
-        
-        // Maintain previous section and previous team while iterating through the recipients
-        // initialize the previous section to "None"
-        String prevSection = Const.DEFAULT_SECTION;
+
         String prevTeam = "";
-        
-        Set<String> sectionsWithResponses = new HashSet<String>();
+
+        String sectionPrefix = String.format("section-%s-", getSectionPosition(section));
+
         Set<String> teamsWithResponses = new HashSet<String>();
-        Set<String> participantsWithResponses = new HashSet<String>();
-          
+        Set<String> teamMembersWithResponses = new HashSet<String>();
+
         // Iterate through the primary participant
         int primaryParticipantIndex = this.getStartIndex();
         for (Entry<String, Map<String, List<FeedbackResponseAttributes>>> primaryToSecondaryParticipantToResponsesMap
                 : sortedResponses.entrySet()) {
             primaryParticipantIndex += 1;
             String primaryParticipantIdentifier = primaryToSecondaryParticipantToResponsesMap.getKey();
-            
+
             String currentTeam = getCurrentTeam(bundle, primaryParticipantIdentifier);
-            String currentSection = getCurrentSection(primaryToSecondaryParticipantToResponsesMap, viewType);
-            
+
+            boolean isStudent = bundle.isParticipantIdentifierStudent(primaryParticipantIdentifier);
+            String participantSection = bundle.getSectionFromRoster(primaryParticipantIdentifier);
+
+            if (isStudent && !participantSection.equals(section)) {
```

Comment:
Which test/sample feedback session were you observing this in? I think we should remove this check, this is the concern of #4486, not here.